doc/examples/denoise_ascm.py

```diff
@@ -0,0 +1,164 @@
+"""
+==============================================================
+Denoise images using Adaptive Soft Coefficient Matching (ASCM)
+==============================================================
+
+Using the non-local means based adaptive denoising [Coupe11]_ you can denoise
+3D or 4D images and boost the SNR of your datasets.
+
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+import nibabel as nib
+from dipy.data import fetch_sherbrooke_3shell
+from dipy.data import read_sherbrooke_3shell
+from dipy.denoise.noise_estimate import estimate_sigma
+from time import time
+from dipy.denoise.non_local_means import non_local_means
+from dipy.denoise.ascm import ascm
+
+"""
+Choose one of the data from the datasets in DIPY
+"""
+
+fetch_sherbrooke_3shell()
+img, gtab = read_sherbrooke_3shell()
+
+data = img.get_data()
+affine = img.get_affine()
+
+mask = data[..., 0] > 80
+data = data[..., 1]
+
+print("vol size", data.shape)
+
+t = time()
```

Comment:
I would remove this timer! You are only using this in line 99. Normally users run code from comments to comments - so this is not informative since it also dependent how fast the user will go through the example.